### PR TITLE
Changed the value provider preference to be backwards compatible

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #3361 

@pranavkm

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/Internal/MvcCoreMvcOptionsSetup.cs [hunk 1]
```diff
@@ -51,9 +51,9 @@ public static void ConfigureMvc(MvcOptions options, IHttpRequestStreamReaderFact
             options.OutputFormatters.Add(new StreamOutputFormatter());
 
             // Set up ValueProviders
+            options.ValueProviderFactories.Add(new FormValueProviderFactory());
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Core/Internal/MvcCoreMvcOptionsSetup.cs]: Could we add a test for this since the ordering is pretty important?

Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Core/Internal/MvcCoreMvcOptionsSetup.cs]: Looks like there is one - https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNet.Mvc.Test/MvcOptionsSetupTest.cs#L71 - shouldn't it fail?

**Comment by ajaybhargavb** [comment]
Updated.

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**